pkg/component/component.go

```diff
@@ -564,51 +572,110 @@ func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo
 }
 
 // ApplyConfigDeleteURL applies url config deletion onto component
-func applyConfigDeleteURL(client *occlient.Client, componentConfig config.LocalConfigInfo) (err error) {
-
-	urlList, err := urlpkg.ListPushed(client, componentConfig.GetName(), componentConfig.GetApplication())
-	if err != nil {
-		return err
-	}
-	localURLList := componentConfig.GetURL()
-	for _, u := range urlList.Items {
-		if !checkIfURLPresentInConfig(localURLList, u.Name) {
-			err = urlpkg.Delete(client, u.Name, componentConfig.GetApplication())
-			if err != nil {
-				return err
+func applyConfigDeleteURL(client *occlient.Client, kClient *kclient.Client, componentConfig config.LocalConfigInfo, envSpecificInfo envinfo.EnvSpecificInfo) (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		// TODO: use envSpecificInfo.GetName()
+		componentName, err := GetComponentName()
+		if err != nil {
+			return err
+		}
+		ingressList, err := urlpkg.ListPushedIngress(kClient, componentName)
+		if err != nil {
+			return err
+		}
+		localURLList := envSpecificInfo.GetURL()
+		for _, i := range ingressList.Items {
+			if !checkIfURLPresentInConfig(checkIfURLPresentInConfigParam{envinfoURL: localURLList, url: i.Name}) {
```

Comment:
Since we are comparing against the name of the URLs, we don't need this function checkIfURLPresentInConfig(). Please create a map from localURLList and then check if the current url name is present in the map or not. That way we can avoid the extra loops in checkIfURLPresentInConfig().